multiqc/modules/rockhopper/rockhopper.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python
+
+""" MultiQC submodule to parse output from Rockhopper summary files
+https://cs.wellesley.edu/~btjaden/Rockhopper/ """
+
+from __future__ import print_function
+import logging
+import re
+import numpy as np
+from collections import OrderedDict
+
+from multiqc import config
+from multiqc.plots import bargraph
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+
+    def __init__(self):
+        
+        # Initialize the parent object
+        super(MultiqcModule, self).__init__(name='Rockhopper', 
+            anchor='rockhopper',
+            href='https://cs.wellesley.edu/~btjaden/Rockhopper/',
+            info="is a comprehensive and user-friendly system "\
+                 "for computational analysis of bacterial RNA-seq data. "\
+                 "It can align reads to genomes, assemble transcripts, "\
+                 "identify transcript boundaries, and discover novel "\
+                 "transcripts such as small RNAs.")
+
+        # Set up vars
+        self.rh_data = dict()
+
+        # Parse summary file
+        for f in self.find_log_files('rockhopper'):
+            self.parse_rockhopper_summary(f)
+         
+        
+        # Filter to strip out ignored sample names
+        self.rh_data = self.ignore_samples(self.rh_data)
+
+        if len(self.rh_data) == 0:
+            raise UserWarning
+        
+        log.info("Found {} reports".format(len(self.rh_data)))
+
+        # Write to file
+        self.write_data_file(self.rh_data, 'multiqc_rockhopper')
+
+        # Basic stats Table
+        self.rockhopper_general_stats_table()
+
+        # Alignment bar plot
+        self.rockhopper_count_bar_plot()
+
+       
+    def parse_rockhopper_summary(self,f):
+       
+        s_name = None
+
+        # Initialize stats fields
+        stats_index = ['mRNA-sense','mRNA-antisense',\
+                       'rRNA-sense','rRNA-antisense',\
+                       'tRNA-sense','tRNA-antisense',\
+                       'ncRNA-sense','ncRNA-antisense','unannotated']
+        
+        results = {name:0 for name in stats_index}
+
+        # Parse rockhopper output line-by-line since there may be many genomes
+        lines = f['f'].split('\n')
+        i = 0 
+        while i < len(lines):
+
+            line = lines[i]
+            
+            # Get sample name
+            if line.startswith('Aligning sequencing reads from file:'):
+                s_name = line.split(':')[1].strip()
```

Comment:
```suggestion
                s_name = line.split(':', 1)[1].strip()
```
Note that I haven't seen what the input files look like, but I guess this change would help for sample names that randomly have `:` characters in them...?

Reply:
Good call